basex-core/src/main/java/org/basex/core/locks/DBLocking.java

```diff
@@ -138,7 +138,9 @@ public void acquire(final Proc pr, final StringList read, final StringList write
       }
       // global read locking
       if(read == null) {
-        while(localWriters > 0) {
+        while(localWriters > 0 &&
+            // We're the only writer, allow global read lock anyway
+            !(1 == localWriters && !(null == write || write.isEmpty()))) {
```

Comment:
Ok, thanks. For my own reference, I'm just listing the cases that are implied by the conditions:

**Loop:**
* `localWriters` > `1`
* `localWriters` = `1`, global write
* `localWriters` = `1`, no local writes

**Stop:**
* `localWriters` = `0`
* `localWriters` = `1`, local writes